task/host-sp-comms/src/main.rs

```diff
@@ -731,6 +748,39 @@ impl ServerImpl {
 
         Ok(())
     }
+
+    /// On success, we will have already filled `self.tx_buf` with our response.
+    /// On failure, our caller should response with
+    /// `SpToHost::KeyLookupResult(err)` with the error we return.
+    fn perform_key_lookup(
+        &mut self,
+        sequence: u64,
+        key: u8,
+        max_response_len: usize,
+    ) -> Result<(), KeyLookupResult> {
+        let key = Key::from_u8(key).ok_or(KeyLookupResult::InvalidKey)?;
+
+        match key {
+            Key::Ping => {
+                const PONG: &[u8] = b"pong";
+
+                if max_response_len < PONG.len() {
+                    return Err(KeyLookupResult::MaxResponseLenTooShort);
+                }
+
+                self.tx_buf.encode_response(
+                    sequence,
+                    &SpToHost::KeyLookupResult(KeyLookupResult::Ok),
+                    |buf| {
+                        buf[..PONG.len()].copy_from_slice(PONG);
```

Comment:
Not specific to this PR, but how do we handle cases where the encoded response is too large for the `tx_buf`?  Is it statically checked anywhere, or would it panic in `encode_response`?

Reply:
It would panic in `encode_response`, yeah. The length of `buf` is statically guaranteed to be at least:

```
use host_sp_messages::{MAX_MESSAGE_SIZE, CHECKSUM_SIZE, Header, SpToHost};

MAX_MESSAGE_SIZE - Header::MAX_SIZE - CHECKSUM_SIZE - SpToHost::MAX_SIZE
```

which is a little less than 4 KiB currently. (The exact size only depends on which `SpToHost` variant we're in; `Header` always serializes to its `MAX_SIZE`.) Clients should either:

a) guarantee they're writing less than this amount of data (which `PONG` technically should, but it's omitted because it's only 4 bytes)
b) limit their writes whatever size `buf` actually is (which is what `get_host_phase2_data` does, the only other message type to date that appends data via this closure)